### ISSUE TITLE
Spaces in path names.....

### ISSUE THREAD

**Issue by Smurf-IV** [NaturalDocs/NaturalDocs#17]
It cannot find paths if they contain spaces !

**Comment by NaturalDocs** [comment]
That's not true at all.  Please post more information.  What is your command line?  Is there a file or folder you have that works when you rename it to not have spaces but doesn't when it does?  What is its name if so?

**Comment by Smurf-IV** [comment]
Structure is 
z:\project\help\NDocs <- Here is the location of the NDoc files
z:\Project\code\source to build\  <- *.sln etc subprojects are subdirecties of this

in th the NDoc project file I had to use ./../../../code/source to build/ <- Yes three back slash pairs .....

and the erro it kept stating was "z:\Project\code\source to build\" cannot be found.

I gave up and uninstalled and I am now investigating DocFX.

**Comment by NaturalDocs** [comment]
If you're still interested in trying, try putting the absolute path in the project file.  It will convert it to a relative path automatically.

I'm not able to reproduce this.  However, if I put in "./../../../code/source to build/" then I get "z:\code\source to build" does not exist, but that's because there's too many ".."s which brings it to z: and then adds on "code\source to build" without "Project".  The correct path would be "..\..\code\source to build", but again, you could have put "z:\project\code\source to build" without worrying about counting the correct number of ".."s.

**Comment by Smurf-IV** [comment]
I did state that it needed "three back slash pairs" to get it to display the "Correct path" in the error message.
I also did try the absolute path, and it just added it onto the end of the NDoc project path location..

I'll give it another go in a few days.
Are there any log files that can be generated to help with this ?
Is there a debug version I can try ?
My OS: Win 8.1 x64

**Comment by NaturalDocs** [comment]
It only generates log files on a crash.  Try doing this:

- Create Project.txt with the absolute path in it and attach it here BEFORE you run Natural Docs
- Run Natural Docs
- Copy and paste the exact console messages it gives you here
- Attach Project.txt here again so I can see what it looks like AFTER you ran Natural Docs.  It will have reformatted itself to use a relative path.

From there I should be able to figure out what was happening.

**Comment by Smurf-IV** [comment]
1. `Source Folder: [..\..\..\Source\Krypton Components]`
2. ![image](https://user-images.githubusercontent.com/2418812/35586286-639db79c-05f2-11e8-85eb-060967412767.png)
3.   ```
      Z:\Krypton-NET-4.7\Help\NDocs>"C:\Program Files (x86)\Natural Docs\NaturalDocs.exe" .
      Natural Docs 2.0.1
      ----------------------
      There were no output targets in Project.txt or the command line.
      There are errors in Z:\Krypton-NET-4.7\Help\NDocs\Project.txt:
         - Line 66: The source folder Z:\Krypton-NET-4.7\Help\Source\Krypton Components] does not exist.
      Run Natural Docs with -h to see the command line reference.
      Z:\Krypton-NET-4.7\Help\NDocs>
```
4. ```
# ERROR: The source folder Z:\Krypton-NET-4.7\Help\Source\Krypton Components] does not exist.
Source Folder: [..\..\..\Source\Krypton Components]
```
So as you can see I needed **3** backslashes for it to get to the the correct path as in the picture.
Here is the message when only 2 are used:
```
# ERROR: The source folder Z:\Krypton-NET-4.7\Help\NDocs\Source\Krypton Components] does not exist.
Source Folder: [..\..\Source\Krypton Components]
```
and an Absolute path looks like this:
```
# ERROR: The source folder Z:\Krypton-NET-4.7\Help\NDocs\[Z\Krypton-NET-4.7\Source\Krypton Components] does not exist.
Source Folder: [Z:\Krypton-NET-4.7\Source\Krypton Components]
```
The rest of the file is just the default one set up from running in a blank directory as per the "Getting Started" instructions

**Comment by NaturalDocs** [comment]
Okay, I see where the confusion is.  You should not be including the brackets in the config file.  So instead of

    Source Folder: [Z:\Krypton-NET-4.7\Source\Krypton Components]

you should be writing just

    Source Folder: Z:\Krypton-NET-4.7\Source\Krypton Components

The parts of the examples in brackets represent things you're supposed to replace with your own values, but the brackets themselves were not supposed to be included.  I'm sorry that that wasn't clear.

**Comment by Smurf-IV** [comment]
Yay, and then Oh
![image](https://user-images.githubusercontent.com/2418812/35693497-480af910-0776-11e8-9e9e-f340623351de.png)
Zip has all of the contents of the NDocs directory (i.e. project files)
[NDocs.zip](https://github.com/NaturalDocs/NaturalDocs/files/1686629/NDocs.zip)

**Comment by NaturalDocs** [comment]
Do that again, but with this version.  It will give me more debugging info on what caused the crash.

http://www.naturaldocs.org/temp/Natural_Docs_temp4.zip

**Comment by Smurf-IV** [comment]
- Copied over the installed version
- Version reported as the same ..
cleaned the output directory so it the same start conditions as before.
- Boom file -> [LastCrash.txt](https://github.com/NaturalDocs/NaturalDocs/files/1690272/LastCrash.txt)

**Comment by NaturalDocs** [comment]
That's not the one at the link.  Running "naturaldocs -v" should show "2.1 (Development Release 02-01-2018)".  You can extract it to any folder, cd into it from a console window, and run it directly if extracting over the installed version doesn't work.

**Comment by Smurf-IV** [comment]
Please re-upload, because I get this ..
![image](https://user-images.githubusercontent.com/2418812/35756883-db0814ac-0864-11e8-9ab5-1f34cf83046f.png)
That time is not my time zone

**Comment by NaturalDocs** [comment]
That's fine.  That information is not updated automatically and I didn't bother to change it for a quick build.  Running with -v shows the internal version number that was changed.  I redownloaded the file at the link and it's correct.

**Comment by SeaVipe** [comment]
I had a minor problem with my original batch file:
@ECHO Off
**C:\Natural Docs\NaturalDocs** ^
-i C:/SRC/TeriBase ^
-i C:/SRC/TeriBase/Modules ^
-i C:/FreeBASIC-1.06.0-win64/inc/CSR ^
-i C:/FreeBASIC-1.06.0-win64/inc/debug ^
--exclude-input C:/SRC/TeriBase/release ^
--exclude-input C:/SRC/TeriBase/release64 ^
-o HTML C:/SRC/TeriBase/Documents ^
-p C:/SRC/TeriBase/Natural Docs ^
--pause-on-error
REM
TIMEOUT 3

Adding double quotes fixed it:
@ECHO Off
**"C:\Natural Docs\NaturalDocs"** ^
-i C:/SRC/TeriBase ^
-i C:/SRC/TeriBase/Modules ^
-i C:/FreeBASIC-1.06.0-win64/inc/CSR ^
-i C:/FreeBASIC-1.06.0-win64/inc/debug ^
--exclude-input C:/SRC/TeriBase/release ^
--exclude-input C:/SRC/TeriBase/release64 ^
-o HTML C:/SRC/TeriBase/Documents ^
-p C:/SRC/TeriBase/Natural Docs ^
--pause-on-error
REM
TIMEOUT 3

**Comment by NaturalDocs** [comment]
I got no response to my last message.  If you're still interested in working on this try again with 2.0.2, as that will generate the same extra debugging information as the zip file I posted earlier.
